exp/internal/controllers/machinepool_controller.go

```diff
@@ -157,18 +157,16 @@ func (r *MachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 		// Always attempt to patch the object and status after each reconciliation.
 		// Patch ObservedGeneration only if the reconciliation completed successfully
-		patchOpts := []patch.Option{}
+		patchOpts := []patch.Option{
```

Comment:
We should always patch owned conditions. Only observedGeneration should only be patched if we didn't get an error.